core/src/io/anuke/mindustry/world/blocks/distribution/MassDriver.java

```diff
@@ -181,6 +181,11 @@ public void drawConfigure(Tile tile){
 
         MassDriverEntity entity = tile.entity();
 
+        entity.waitingShooters.forEach(shooter -> {
```

Comment:
forEach isn't available on Android. Use a standard for-loop. In general, standard for-loops are preferable unless you are doing a one-liner (e.g. `array.each(e -> e.x = y)` would be a valid use case.)

Reply:
really? so much for java being cross platform, i'll get this fixed 😋

Reply:
"Write once, run anywhere" is basically a lie; once your program becomes more complex than a simple command line reader you have to start dealing with many platform-specific issues.

Reply:
Well personally i would expect to run into those kind of things when trying to make the device vibrate (sounds wrong, but you get the point), not with something as "simple" as a foreach :D

Reply:
curious, isn't it? you'd think that android would come with the standard set of java se commands, seeing as the android api is built on the language.

Reply:
It might have been built on the same foundation, but many devices are old and don't support these ""new"" Java 8 features. It's not that forEach is completely unsupported; it just requires a newer API level than Mindustry is supposed to support.